### ISSUE TITLE
[Bug]: Deadlock on notes_data

### ISSUE THREAD

**Issue by dpfaffenbauer** [pimcore/pimcore#15064]
### Pimcore version

10.x or 11.x

### Steps to reproduce

On high concurrency systems where lot of notes are written, the table ends up locking all the time. Reason is cause it uses id and name is primary index. If we change the index to be auto-incremented, it doesn't deadlock anymore.

Current Table is:

```
CREATE TABLE `notes_data` (
  `id` int(11) NOT NULL,
  `name` varchar(255) NOT NULL,
  `type` enum('text','date','document','asset','object','bool') DEFAULT NULL,
  `data` text,
  PRIMARY KEY (`id`, `name`)
) DEFAULT CHARSET=utf8mb4;
```

### Expected / actual behavior

Better Layout would be:

```
CREATE TABLE `notes_data` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `cid` int(11) NOT NULL,
  `name` varchar(255) NOT NULL,
  `type` enum('text','date','document','asset','object','bool') DEFAULT NULL,
  `data` text DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `id_cid` (`id`,`cid`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;

```

**Comment by dpfaffenbauer** [comment]
Chaning that for Pimcore 10 without a BC break is quite hard I think, not sure if you wanna have a BC break here for Pimcore 11?

**Comment by brusch** [comment]
Why would this be a breaking change?

**Comment by dpfaffenbauer** [comment]
getId would not return the "Content ID" anymore, but the Auto Incremented it.

**Comment by brusch** [comment]
Ah yeah, but we could keep the schema compatible by just adding another auto-increment column and do this with a minor release. 
I'm afraid for 11.0 it's too late now 😬

**Comment by dpfaffenbauer** [comment]
@brusch Interestingly, the model has fields for cid and ctype.

**Comment by dpfaffenbauer** [comment]
nevermind, I didn't realize it is split up into notes and notes_data. I will create a pr for the auto_inc

**Comment by SamyMP** [comment]
Hi, 

For your information, we are facing the exact same problem with the `object_meatada` table on one of our object. 

What happens is, we are trying to populate a fresh install of pimcore with a little bit more than 100 000 DataObjects through the DataImporter Bundle. These DataObjects have multiple relation fields. Some of these fields are filled through the DataImporter, but others are filled using an EventListener (using the `PreSaveEvent` of the DataImporter) because what we are trying to achieve is not possible natively with the DataImporter.

When we execute the import in parallel with multiple process (ie. 10), a lot of objects couldn't be created because of this error :
```
Next Doctrine\DBAL\Exception\DeadlockException: An exception occurred while executing 'INSERT INTO object_metadata_OBJECT (o_id, dest_id, fieldname, ownertype, ownername, index, position, type, column, data) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE o_id = ?, dest_id = ?, fieldname = ?, ownertype = ?, ownername = ?, index = ?, position = ?, type = ?, column = ?, data = ?;' with params [my params]:

SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction in /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:51
```

It failed even though it retried, according to the message.

The params (my enveloppes) are good since, to try to debug this, we imported all of our objects by executing the parallel process with 1 process (took 6 hours, but no errors). It is clearly linked to the parallelization of the import.

I also have to mention that we were able to import everything by commenting our event in the DataImporter, with multiple processes (again, 10). So the events in the DataImporter seems to be a part of the problem as well in our case. 

Lastly, the problem doesn't seem to appear when updating already created objects. It only affects the creation.

Should I create a new issue with this problem, or is it also part of this problem ?

**Comment by mcop1** [comment]
@SamyMP Please create a new issue for that

**Comment by brusch** [comment]
@SamyMP or even better a PR, shouldn't be a big deal 😉

**Comment by SamyMP** [comment]
Unfortunately, I won't have time to investigate the problem myself, I'll let you know if I can give it a shot but I'd rather create an issue first, in case you have a clear idea of what has to be done 🙂

**Comment by dpfaffenbauer** [comment]
It seems like the same issue, it locks the whole table to insert the primary key, if we add a auto_id, it doesn't need to lock the whole table.

**Comment by brusch** [comment]
Yep, I'd assume so as well.

**Comment by dpfaffenbauer** [comment]
This still happens on high concurrent systems, I guess the issue here is the same as with relations, pimcore just deletes the data and inserts it again. this is locking the notes_data table.
